lib/HLSL/DxilValidation.cpp

```diff
@@ -708,8 +708,15 @@ struct ValidationContext {
     BasicBlock *BB = I->getParent();
     Function *F = BB->getParent();
 
-    DiagPrinter << "at " << I;
-    DiagPrinter << " inside block ";
+    std::string InstrStr;
+    raw_string_ostream InstrStream(InstrStr);
+    I->print(InstrStream);
+    InstrStream.flush();
+    StringRef InstrStrRef = InstrStr;
+    InstrStrRef = InstrStrRef.ltrim(); // Ignore indentation
+    DiagPrinter << "at '" << InstrStrRef << "'";
```

Comment:
I could also have added one more overload of `operator<<` but I didn't see a point to it.

Reply:
Below, the function name is printed when `<< *F` is used, since this goes through the `const Value &` overload.  That might have been slightly better than going down the `const void*` path for the instruction pointer, but this is better.  I'm not sure using a new overload would be the right path here, because you want to print the entire instruction in this case, not just the name if it has one, and you probably don't want to change the default behavior for values when they are instructions.

Reply:
By the way, is the whole point of using a new stream here, rather than printing directly to the DiagStream(), to remove the indentation from the instruction print?

Reply:
I think I simply hadn't thought of it, but it seems like the only way to strip the indentation, now that you say so.